lib/chore/strategies/consumer/threaded_consumer_strategy.rb

```diff
@@ -56,14 +56,14 @@ def start_consumer_thread(queue)
         t = Thread.new(queue) do |tQueue|
           begin
             consumer = Chore.config.consumer.new(tQueue)
-            consumer.consume do |message_id, message_receipt_handle, queue_name, queue_timeout, body, previous_attempts|
+            consumer.consume do |message_id, message_receipt_handle, queue_name, queue_timeout, body, previous_attempts, received_timestamp|
               # Quick hack to force this thread to end it's work
               # if we're shutting down. Could be delayed due to the
               # weird sometimes-blocking nature of SQS.
               consumer.stop if !running?
               Chore.logger.debug { "Got message: #{message_id}"}
 
-              work = UnitOfWork.new(message_id, message_receipt_handle, queue_name, queue_timeout, body, previous_attempts, consumer)
+              work = UnitOfWork.new(message_id, message_receipt_handle, queue_name, queue_timeout, body, previous_attempts, consumer, nil, nil, received_timestamp)
```

Comment:
New Implementation 5/20: Called from `consumer.rb`, takes in new `received_timestamp` param and passes it in to create UnitOfWork. (which will then be used by `worker.rb` for the worker duplicate handling)

(2/5)